src/AdTypes.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\fb_instant_articles_display\DrupalInstantArticleDisplay.
+ */
+
+namespace Drupal\fb_instant_articles;
+
+abstract class AdTypes {
```

Comment:
@scottrigby I was actually just trying to implement a simple enum and took a cue from the comments on the [`SplEnum` docs](http://php.net/manual/en/class.splenum.php). I liked having something that was statically replaced at compile time, so it was not possible to have an undiscovered typo in there.

That being said, something dynamic like your approach could mitigate alleviate the issues I mentioned in the message for b52ca3350f8076fe0dab4d49bad3f9ca09902803

I'll try to implement something like that instead.